Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -61,17 +62,18 @@ public Color Tint
 
 		internal Task CurrentNavigationTask { get; set; }
 
-		Stack<Page> INavigationPageController.StackCopy
+		Page INavigationPageController.Peek(int depth)
 		{
-			get
+			if (PageController.InternalChildren.Count <= depth)
 			{
-				var result = new Stack<Page>(PageController.InternalChildren.Count);
-				foreach (Page page in PageController.InternalChildren)
-					result.Push(page);
-				return result;
+				return null;
 			}
+
+			return (Page)PageController.InternalChildren[PageController.InternalChildren.Count - depth - 1];
 		}
```

Comment:
Peek(-1) would be a bad thing, right?  Should probably return null if depth < 0?